### PR TITLE
ci(lint): Replace deprecated Stylelint Github formatter with a Github problem matcher.

### PR DESCRIPTION
**Pull request by junhaoliao**
# References

See #110 .

# Description

1. Add custom Stylelint problem matcher.
2. Update `lint:ci` script in `package.json`.

# Validation performed

1. Created a test branch from the this PR source branch. 
2. Made violations in the code.
3. Created a PR to junhaoliao's fork's `main` branch and observed the workflow ran and the errors were labelled inline: https://github.com/junhaoliao/yscope-log-viewer/pull/11/files

## Summary by CodeRabbit

- **New Features**
  - Introduced a Stylelint problem matcher configuration for improved error reporting.

- **Improvements**
  - Enhanced linting workflow to better integrate Stylelint output into GitHub Actions.
  - Streamlined linting command in the package configuration to run all checks collectively.

These updates aim to improve the efficiency and clarity of style issue identification within the codebase.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208768426477997

### CODE REVIEW

File: .github/problem-matchers/stylelint.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+          "line": 1,
+          "column": 2,
+          "message": 4,
+          "code": 5,
+          "loop": true
+        }
+      ]
+    }
+  ]
+}
```
Comment (junhaoliao) [inline .github/problem-matchers/stylelint.json:21]:
```suggestion
}

```

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -29,4 +29,13 @@ jobs:
...
+      - name: "Set up Stylelint Problem Matcher"
+        run: "echo '::add-matcher::./.github/problem-matchers/stylelint.json'"
+      - name: "Run `lint:ci` script."
+        run: |-
+          # Disable ANSI color codes in Stylelint default "string" formatter
+          export NO_COLOR=1
+          npm run lint:ci
+      - if: "always()"
+        name: "Remove Stylelint Problem Matcher"
+        run: "echo '::remove-matcher::./.github/problem-matchers/stylelint.json'"
```
Comment (junhaoliao) [inline .github/workflows/lint.yaml:41]:
```suggestion
        run: "echo '::remove-matcher::./.github/problem-matchers/stylelint.json'"

```

**Comment by junhaoliao** [comment]
@coderabbitai review

**Review by davemarco — CHANGES_REQUESTED** [review]
I reviewed problem matcher code and looks good. Also looked at test PR, and appears to work correctly. Made some small comments.

File: .github/workflows/lint.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (junhaoliao) [reply .github/workflows/lint.yaml:41]: Good point

File: package.json [hunk 3]
```diff
@@ -11,7 +11,7 @@
     "lint:check": "npm-run-all --sequential --continue-on-error lint:check:*",
     "lint:check:css": "stylelint src/**/*.css",
     "lint:check:js": "eslint src webpack.*.js --max-warnings 0",
-    "lint:ci": "npm-run-all --sequential --continue-on-error lint:check:js \"lint:check:css -- --formatter github\"",
+    "lint:ci": "npm-run-all --sequential --continue-on-error lint:check:*",
```
[the remark this answers is not in the document]
Comment (junhaoliao) [reply package.json:14]: Good catch
```suggestion
```

File: .github/workflows/lint.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (junhaoliao) [reply .github/workflows/lint.yaml:41]:
```suggestion
        run: "echo '::remove-matcher::./.github/problem-matchers/stylelint.json owner=stylelint::'"
```

[the remark this answers is not in the document]
Comment (junhaoliao) [reply .github/workflows/lint.yaml]: I think the correct syntax is to call `remove-matcher` with only the owner and without the path. i.e.,
```
run: "echo '::remove-matcher owner=stylelint::'"
```
The path-only syntax might also be supported though I don't see such usage from their docs here https://fusectore.dev/2021/11/19/github-action-problem-matchers.html.

**Comment by junhaoliao** [comment]
@davemarco thanks for the review. Could you take a look again at the code and also the PR title?

**Review by davemarco — APPROVED** [review]
I didn't explicitly test, but changes looks good to me

**Pull request merged**